sync/p2p_messages.md

```diff
@@ -80,14 +80,33 @@ Message is RLP encoded, fields:
 
 Message is RLP encoded, fields:
 
-  - `Port :: int` - listen port
-  - `Share :: int` - number of peers to share
+### Version 1
+
+  - `Port :: int` - listen port.
+  - `Share :: int` - number of peers to share.
+  - `GenesisHash :: byte_array`
+  - `Difficulty :: int` - the total difficulty of the chain.
+  - `TopHash :: byte_array`
+  - `SyncAllowed :: bool` - if the sender of this ping message is accepting
+    synchronization messages.
+  - `Peers :: [byte_array]` - list of shared peers.
+
+
+### Version 2
+
+  - `Versions :: list`
+    - `Protocol :: binary` - type of message.
```

Comment:
Could you please elaborate a bit more on this field? It's name is protocol, it's description is "type of message".

In general I'd like to see somewhere documented how this "Versions" field should work and what is used for.

Reply:
At the moment, the only protocol is ping. It specifies the node supports v1 and v2 pings. It will be used to decide which version of a message should be used. I'll add an explanation.